storage/sealer/fsutil/statfs_unix.go

```diff
@@ -15,12 +20,113 @@ func Statfs(path string) (FsStat, error) {
 		return FsStat{}, xerrors.Errorf("statfs: %w", err)
 	}
 
+	used := int64(stat.Blocks-stat.Bavail) * int64(stat.Bsize)
+	available := int64(stat.Bavail) * int64(stat.Bsize)
+
+	// Filesystems like ZFS can contain sub-volumes which are distinct filesystems, but use the same pool of space.
+	// To accurately account for space used in sub-volumes, accumulate the used space in each filesystem mounted under the path.
+	mounts, err := getMountsUnder(path)
+	if err != nil {
+		log.Warnf("could not get all mounts under %s: %e", path, err)
+	}
+
+	for _, mount := range mounts {
+		var mountStat syscall.Statfs_t
+		if err := syscall.Statfs(mount, &mountStat); err != nil {
+			log.Warnf("could not get stats for mount %s: %e", mount, err)
+			continue
+		}
+		used += int64(mountStat.Blocks-mountStat.Bavail) * int64(mountStat.Bsize)
+	}
+
 	// force int64 to handle platform specific differences
 	//nolint:unconvert
 	return FsStat{
-		Capacity: int64(stat.Blocks) * int64(stat.Bsize),
+		Capacity: used + available,
```

Comment:
I'm actually not sure if this correct, the capacity we care about here is really "how much data can I put in this path in total", so it maybe makes sense that it drops if it's used in another subvolume.

Afaict we're only using the `Capacity` field in the CLI, which I guess can break the display in some weird ways?

Reply:
It is primarily an interface issue I was aiming to fix. As is, when the sealed subvolume grows, instead of showing more drive space used, the total size of the drive gradually goes down.